temporian/core/evaluator.py

```diff
@@ -26,26 +26,50 @@
 
 AvailableBackends = Any
 Data = Dict[Event, Union[str, pathlib.Path, pandas_event.PandasEvent]]
-Query = Union[Event, List[Event], Set[Event]]
+Query = Union[Event, List[Event], Set[Event], Dict[str, Event]]
 
 
 def evaluate(
     query: Query,
     input_data: Data,
-    backend: AvailableBackends = "pandas",
+    backend: AvailableBackends = "numpy",
 ) -> Dict[Event, Any]:
-    """Evaluates a query on data."""
+    """Evaluates a query on data.
 
-    # Normalize query
+    Args:
+        query: Events to compute. Support event, dict and list of events.
+        input_data: Dictionary of event and event values to use for the
+          computation.
+        backend: Type of backend to use.
+
+    Returns:
+        An object with the same structure as "event" containing the results. For
+        instance, if "event" is a dictionary of events, the returned object
+        will be a dictionary of event results. If "event" is a list of events,
+        the returned value will be a list of event values with the same order.
+    """
+
+    # Normalize query into a list
     normalized_query: List[Event] = {}
+
+    Q_TYPE_DICT = "DICT"
+    Q_TYPE_LIST = "LIST"
+    Q_TYPE_VALUE = "VALUE"
+
     if isinstance(query, Event):
+        # The query is a single value
         normalized_query = [query]
-
-    elif isinstance(query, set):
-        normalized_query = list(query)
+        query_type = Q_TYPE_VALUE
```

Comment:
Shouldn't `Q_TYPE_VALUE` be for the first case (`query` is an `Event`) and this be `Q_TYPE_SET`?